go-eicio/writer.go

```diff
@@ -15,11 +17,23 @@ func Create(filename string) (*Writer, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if strings.HasSuffix(filename, ".gz") {
+		return NewGzipWriter(file)
```

Comment:
calling `Close()` on the returned `eicio.Writer` won't close the underlying file as per the documentation of `gzip.Writer`:

```
$> go doc compress/gzip.Writer.Close 
func (z *Writer) Close() error
    Close closes the Writer, flushing any unwritten data to the underlying
    io.Writer, but does not close the underlying io.Writer.
```

so you're leaking resources (file handles.)

I would recommand instead just exposing `NewWriter` and handling gzip streams like so:

```go
func main() {
    f, err := os.Create(fname)
    defer f.Close()
    var w io.Writer = f
    if doGZip {
        gw, err := gzip.NewWriterLevel(w, gzip.BestSpeed)
        defer gw.Close()
        w = gw
    }
    ew, err := eicio.NewWriter(w)
    defer ew.Close()
    // ...
    if err = ew.Close(); err != nil { ... }
    if err = f.Close(); err != nil { ... }
}
```